crates/tx5-go-pion/src/evt.rs

```diff
@@ -56,150 +59,444 @@ pub enum PeerConnectionEvent {
-                    MANAGER.lock().peer_con.get(&peer_con_id).cloned();
-                if let Some(cb) = maybe_cb {
-                    cb(PeerConnectionEvent::DataChannel(DataChannelSeed::new(
-                        data_chan_id,
-                    )));
-                }
+                manager_access!(
+                    MANAGER,
+                    peer_con_id,
+                    offload,
+                    peer_con,
+                    async {
+                        let recv_limit = match peer_con.get_recv_limit() {
+                            Ok(recv_limit) => recv_limit,
+                            Err(err) => {
+                                API.data_chan_free(data_chan_id);
+                                return Err(err);
+                            }
+                        };
+
+                        let (chan, recv) =
+                            DataChannel::new(data_chan_id, recv_limit);
+
+                        peer_con
+                            .send_evt(PeerConnectionEvent::DataChannel(
+                                chan, recv,
+                            ))
+                            .await
+                    }
+                );
             }
             SysEvent::DataChanClose(data_chan_id) => {
-                let maybe_cb =
-                    MANAGER.lock().data_chan.get(&data_chan_id).cloned();
-                if let Some(cb) = maybe_cb {
-                    cb(DataChannelEvent::Close);
-                }
+                manager_access!(
+                    MANAGER,
+                    data_chan_id,
+                    offload,
+                    data_chan,
+                    data_chan.send_evt(DataChannelEvent::Close)
+                );
             }
             SysEvent::DataChanOpen(data_chan_id) => {
-                let maybe_cb =
-                    MANAGER.lock().data_chan.get(&data_chan_id).cloned();
-                if let Some(cb) = maybe_cb {
-                    cb(DataChannelEvent::Open);
-                }
+                manager_access!(
+                    MANAGER,
+                    data_chan_id,
+                    offload,
+                    data_chan,
+                    data_chan.send_evt(DataChannelEvent::Open)
+                );
             }
             SysEvent::DataChanMessage {
                 data_chan_id,
                 buffer_id,
             } => {
-                let maybe_cb =
-                    MANAGER.lock().data_chan.get(&data_chan_id).cloned();
-                if let Some(cb) = maybe_cb {
-                    cb(DataChannelEvent::Message(GoBuf(buffer_id)));
-                }
+                let mut buf = GoBuf(buffer_id);
+                manager_access!(
+                    MANAGER,
+                    data_chan_id,
+                    offload,
+                    data_chan,
+                    async {
+                        let len = buf.len()?;
+                        if len > 16 * 1024 {
```

Comment:
Is this the same as the EventSend capacity, or something different? Names and/or intentions would be great for these constants.

Reply:
This is a core limit of the webrtc protocol itself, so there's no need to test it... but I can turn it into a constant at least.

Reply:
done: https://github.com/holochain/tx5/pull/77/commits/430e250b4604163d5ea183b15e5ba10919cd6f49